src/screens/Game/ReadingMain.tsx

```diff
@@ -38,39 +40,40 @@ const TOTAL_TIME = gameDescriptions.Reading.minutes * 60;
 type Props = NativeStackScreenProps<RootStackParamList, "ReadingMain">;
 
 export default function ReadingMain({ navigation, route }: Props) {
-  const [storyArray, setStoryArray] = useState(getStoryArray());
-  const [page, setPage] = useState(0);
+  const dispatch = useDispatch();
 
-  /**
-   * This function will pull up the next paragraph of the current article or move on
-   * to the next one once the user is finished with this article.
-   */
-  const nextParagraph = () => {
-    if (storyArray.length - 1 === page) {
-      setStoryArray(getStoryArray());
-      setPage(0);
-    } else {
-      setPage(page + 1);
-    }
-  };
+  const { paragraph, nextParagraph, onTimeComplete } = useReadingProblems({
+    navigation,
+    route,
+  });
 
-  const nextArticle = () => {
-    setStoryArray(getStoryArray());
-    setPage(0);
+  const nextSection = () => {
+    Alert.alert(
+      "Skip Reading Section",
+      "Are you sure you want to skip the Reading section?",
+      [
+        {
+          text: "Cancel",
+          style: "cancel",
+        },
```

Comment:
We would not want the timer to continue counting down while the confirmation alert is shown. Pause the timer before the alert and unpause when cancel is clicked